src/pages/SubscriptionService.tsx

```diff
@@ -67,6 +68,21 @@ class SubscriptionService extends Component<Props, SubscriptionServiceState> {
     }
   };
 
+  public handleSelectChange = (event: ChangeEvent<HTMLSelectElement>) => {
+    const target = event.target;
+    const value = target.value;
+    const name = target.name;
+    const serviceDecoding = Service.decode({
+      ...this.state.service,
+      [name]: inputValueMap(name, value)
+    });
+    if (serviceDecoding.isRight()) {
+      this.setState({
+        service: serviceDecoding.value
+      });
+    }
+  };
+
```

Comment:
https://github.com/pagopa/io-handbook/blob/master/development-guidelines.md#general-guidelines
> Whenever and wherever possible, prefer functional style over imperative and immutable data structure over mutable state.

Although this case is very simple and there's no much improvement, try to avoid the use of `if/else` branches and `.value` when using a monadic type (like `Either` in this case). Instead, `fp-ts` offers a composable api to deal with such types. It's a bit counter-intuitive at the beginning, but it has huge benefit in the long run.

```suggestion
  public handleSelectChange = (event: ChangeEvent<HTMLSelectElement>) => {
    const { target: { name, value } } = event;
    Service.decode({
      ...this.state.service,
      [name]: inputValueMap(name, value)
    }).chain(service => this.setState({ service }))
  };
```

Reply:
More: what happens when `Service.decode` fails? Is there a case in which you think it can happen? Should we handel it in some way (a log for example)?

Reply:
This code is triggered when the user is typing a field and so `Service.decode`  may fail and nothing is done.
Could I add an error message shown to the user?
Now, however, there are no error messages shown to the user.

Reply:
but how can I use `Either` with `chain `if the return type of `this.setState({service}) `is void?

Reply:
Yes, my bad. `map` is actually the right operation, not `chain`.